auth/server/src/main/java/com/seagates3/dao/ldap/UserPolicyLdapStore.java

```diff
@@ -0,0 +1,99 @@
+package com.seagates3.dao.ldap;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.novell.ldap.LDAPAttribute;
+import com.novell.ldap.LDAPException;
+import com.novell.ldap.LDAPModification;
+import com.seagates3.exception.DataAccessException;
+import com.seagates3.model.Policy;
+import com.seagates3.model.User;
+import com.seagates3.model.UserPolicy;
+
+public
+class UserPolicyLdapStore {
+
+ private
+  final Logger LOGGER =
+      LoggerFactory.getLogger(UserPolicyLdapStore.class.getName());
+
+ public
+  void attach(Object userPolicyObj) throws DataAccessException {
+    UserPolicy userPolicy = (UserPolicy)userPolicyObj;
+    User user = userPolicy.getUser();
+    Policy policy = userPolicy.getPolicy();
+
+    ArrayList modList = new ArrayList();
+    LDAPAttribute attr;
+
+    String dn =
+        String.format("%s=%s,%s=%s,%s=%s,%s=%s,%s", LDAPUtils.USER_ID,
+                      user.getId(), LDAPUtils.ORGANIZATIONAL_UNIT_NAME,
+                      LDAPUtils.USER_OU, LDAPUtils.ORGANIZATIONAL_NAME,
+                      user.getAccountName(), LDAPUtils.ORGANIZATIONAL_UNIT_NAME,
+                      LDAPUtils.ACCOUNT_OU, LDAPUtils.BASE_DN);
+
+    if (policy.getPolicyId() != null) {
+      List<String> userPolicyIds = user.getPolicyIds();
+      userPolicyIds.add(policy.getPolicyId());
+      String[] policyIds = new String[userPolicyIds.size()];
```

Comment:
Need to increment attachment_count for policy when you are attaching a policy to a user.
Plz handle it.
Exactly opposite for detach- decrement the counter

Reply:
Done